libs/component-library/src/lib/theme/theme.ts

```diff
@@ -93,111 +93,149 @@ const fxLightTheme = createTheme({
+};
+
+const textVariants = {
+  defaults: {
+    color: 'content1',
+  },
+  body: {
+    fontSize: 16,
+  },
+  eyebrow2: {
+    fontFamily: 'Montserrat-Medium',
+    fontSize: 8,
+    lineHeight: 10,
+    textTransform: 'uppercase',
+  },
+  bodyXXSRegular: {
+    fontFamily: 'OpenSans-Regular',
+    fontSize: 10,
+    lineHeight: 14,
+  },
+  bodyXSLight: {
+    fontFamily: 'OpenSans-Light',
+    fontSize: 12,
+    lineHeight: 16,
+  },
+  bodyXSRegular: {
+    fontFamily: 'OpenSans-Regular',
+    fontSize: 12,
+    lineHeight: 16,
+  },
+  bodyXSSemibold: {
+    fontFamily: 'OpenSans-Semibold',
+    fontSize: 12,
+    lineHeight: 16,
+  },
+  bodySmallLight: {
+    fontFamily: 'OpenSans-Light',
+    fontSize: 14,
+    lineHeight: 24,
+  },
+  bodySmallRegular: {
+    fontFamily: 'OpenSans-Regular',
+    fontSize: 14,
+    lineHeight: 24,
+  },
+  bodySmallSemibold: {
+    fontFamily: 'OpenSans-Semibold',
+    fontSize: 14,
+    lineHeight: 24,
+  },
+  bodyLargeLight: {
+    fontFamily: 'OpenSans-Light',
+    fontSize: 20,
+    lineHeight: 30,
+  },
+  bodyLargeRegular: {
+    fontFamily: 'OpenSans-Regular',
+    fontSize: 20,
+    lineHeight: 30,
+  },
+  bodyMediumLight: {
+    fontFamily: 'OpenSans-Light',
+    fontSize: 16,
+    lineHeight: 28,
+  },
+  bodyMediumRegular: {
+    fontFamily: 'OpenSans-Regular',
+    fontSize: 16,
+    lineHeight: 28,
+  },
+  h200: {
+    fontFamily: 'Montserrat-SemiBold',
+    fontSize: 18,
+    lineHeight: 24,
+  },
+  h300: {
+    fontFamily: 'Montserrat-Semibold',
+    fontSize: 24,
+    lineHeight: 36,
+  },
+  h400: {
+    fontFamily: 'Montserrat-Semibold',
+    fontSize: 28,
+    lineHeight: 40,
+  },
+};
+
+const buttonTheme = {
+  container: {
+    types: {
+      defaults: {
+        backgroundColor: 'greenBase',
+        borderRadius: 's',
+        alignItems: 'center',
+        justifyContent: 'center',
+      },
+      inverted: {
+        backgroundColor: undefined,
+        borderColor: 'primary',
+        borderWidth: 2,
+      },
+      disabled: {
+        backgroundColor: 'backgroundSecondary',
+      },
+      pressed: {
+        backgroundColor: 'greenPressed',
+      },
     },
-    inverted: {
-      backgroundColor: undefined,
-      borderColor: 'primary',
-      borderWidth: 2,
+    sizes: {
+      defaults: {
+        height: 40,
+      },
+      large: {
+        height: 60,
+      },
     },
-    disabled: {
-      backgroundColor: 'backgroundSecondary',
+  },
+  text: {
+    types: {
+      defaults: { color: 'white' },
+      inverted: { color: 'greenBase' },
+      disabled: { color: 'border' },
+      pressed: { color: 'white' },
     },
-    pressed: {
-      backgroundColor: 'greenPressed',
+    sizes: {
+      defaults: { ...textVariants.bodyXSSemibold },
+      large: { ...textVariants.bodySmallSemibold },
     },
   },
+};
+
+const fxLightTheme = createTheme({
+  ...BaseTheme,
+  textVariants,
+  buttonVariants: { ...buttonTheme.container.types },
```

Comment:
with `buttonVariants`, do we want them to be able to pass in `disabled` and `pressed` when creating a button?

Reply:
hmm, looks like the old implementation allowed these 2 options to be used as `buttonClasses` before

Reply:
Yeah the old version did allow it. Personally I don't think it's necessary to allow but doesn't hurt leaving it in